h5py/_hl/dataset.py

```diff
@@ -251,10 +258,19 @@ def __len__(self):
         """
         return len(self._dset)
 
-    def __array__(self):
+    def __array__(self, dtype=None, copy=True):
+        if dtype not in (None, object):
+            raise TypeError(
+                "AsStrWrapper.__array__ doesn't support the dtype argument"
+            )
+        if copy not in (True, _NUMPY_COPY_IF_NEEDED):
+            raise ValueError(
+                f"AsStrWrapper.__array__ received {copy=} "
+                f"but memory allocation cannot be avoided on read"
+            )
         return numpy.array([
             b.decode(self.encoding, self.errors) for b in self._dset
-        ], dtype=object).reshape(self._dset.shape)
+        ], dtype=object, copy=_NUMPY_COPY_IF_NEEDED).reshape(self._dset.shape)
```

Comment:
```suggestion
        ], dtype=object).reshape(self._dset.shape)
```

We're passing in a list, so Numpy is always creating a new (object) array here in any case. So we may as well not clutter the code up.